plugin/src/main/java/net/dzikoysk/funnyguilds/listener/region/PlayerMove.java

```diff
@@ -34,14 +34,15 @@ public void onMove(PlayerMoveEvent event) {
         Location to = event.getTo();
         Player player = event.getPlayer();
 
+        if (to == null) {
+            return;
+        }
+
+        if (from.getBlockX() == to.getBlockX() && from.getBlockZ() == to.getBlockZ()) {
+            return;
+        }
+        
         Bukkit.getScheduler().runTaskAsynchronously(this.plugin, () -> {
-            if (to == null) {
-                return;
-            }
-
-            if (from.getBlockX() == to.getBlockX() && from.getBlockZ() == to.getBlockZ()) {
-                return;
-            }
 
             Option<User> userOption = this.userManager.findByPlayer(player);
```

Comment:
Można jeszcze wystawić tego return'a przed taskiem.

Reply:
https://github.com/FunnyGuilds/FunnyGuilds/blob/5461a2a55b5a626167386a4af2ceb01b0fc87012/plugin/src/main/java/net/dzikoysk/funnyguilds/listener/region/PlayerMove.java#L47

It performs operations on memory which takes CPU time and performing it synchronously on the server thread would unnecessarily block the main server thread, so what you wrote makes no sense. The operations moved above are the result of the data received in the event.